MSVC compile error: MolHash scoped enum cannot be redeclared as unscoped

In preparation for updating the conda-forge recipe, I just attempted to build RDKit on Windows using the Visual Studio compiler version that is used by conda-forge on Azure Pipelines (cmake says MSVC 19.0.24234.1, I believe this is from a VS2017 install with optional extra component VC++ 2015.3 v14).

I get the following error from the new MolHash code:

```
c:\anaconda\conda-bld\debug_1572451087860\work\code\graphmol\molhash\nmmolhash.h(45): error C3431: 'HashFunction': a scoped enumeration cannot be redeclared as an unscoped enumeration
```

The RDKit CI on Azure Pipelines (apparently using MSVC 19.16.27034.0) does not seem to have this problem, so I guess it is specific to the older version of the compiler. 

Hopefully we could work around this with a patch in the conda-forge recipe? But I haven't been able to figure out how yet.

**Configuration:**

- RDKit Version: 2019.09.1
- Operating system: Windows

--- a/Code/GraphMol/MolHash/hashfunctions.cpp
+++ b/Code/GraphMol/MolHash/hashfunctions.cpp
@@ -680,7 +680,7 @@ static std::string ArthorSubOrderHash(RWMol *mol) {
   return buffer;
 }
 
-std::string MolHash(RWMol *mol, enum HashFunction func) {
+std::string MolHash(RWMol *mol, HashFunction func) {
   std::string result;
   char buffer[32];
   NMRDKitSanitizeHydrogens(mol);

--- a/Code/GraphMol/MolHash/nmmolhash.h
+++ b/Code/GraphMol/MolHash/nmmolhash.h
@@ -42,7 +42,7 @@ enum class RDKIT_MOLHASH_EXPORT HashFunction {
   ArthorSubstructureOrder = 17
 };
 
-RDKIT_MOLHASH_EXPORT std::string MolHash(RWMol *mol, enum HashFunction func);
+RDKIT_MOLHASH_EXPORT std::string MolHash(RWMol *mol, HashFunction func);
 
 enum class RDKIT_MOLHASH_EXPORT StripType {
   AtomStereo = 1,

